lib/bundler/fetcher.rb

```diff
@@ -154,37 +130,26 @@ def fetch_spec(spec)
       elsif cached_spec_path = gemspec_cached_path(spec_file_name)
         Bundler.load_gemspec(cached_spec_path)
       else
-        Bundler.load_marshal Gem.inflate(fetch(uri))
+        Bundler.load_marshal Gem.inflate(fetchers.first.fetch uri)
```

Comment:
Seems like maybe `fetch` should be a class method if it literally doesn't matter which fetcher instance it uses?